### DISCUSSION TITLE
Session token expire in Polaris Axios call

### DISCUSSION THREAD

**Discussion by muhammadasfar** [osiset/laravel-shopify#978] [Q&A]
Did anyone develop an app using this package and Polaris?

Currently, I am facing an issue with session token (session token expire) in the axios call.

**Comment by extfoxDusan** [comment]
I am trying to build a custom AJAX Call to the controllers, and it's not recognized.

**Comment by sp-artisan** [comment]
Hello, 

I was having same issue in my windows machine, I solved it by time sync in my computer I hope it helps you.

**Comment by Kyon147** [comment]
If you are using axios as well, you want to use the interceptor functionality.

This way you can set up a function to run before any ajax call that gets the session token to make sure you always get a valid one before making any request.

Here's the interceptor code I've made for this package 

````
axios.interceptors.request.use(function (config) {
    return utils.getSessionToken(window['app-bridge-app']) // requires a Shopify App Bridge instance
        .then((token) => {
            // console.log('axios token')
            // Append your request headers with an authenticated token
            config.headers.Authorization = `Bearer ${token}`
            return config
        })
})
````
